GoogleSignIn/Sources/GIDAppCheck/Implementations/GIDAppCheck.m

```diff
@@ -52,45 +56,64 @@ - (BOOL)isPrepared {
   return [self.userDefaults boolForKey:kGIDAppCheckPreparedKey];
 }
 
-- (void)prepareForAppCheckWithCompletion:(nullable void (^)(FIRAppCheckToken * _Nullable,
-                                                            NSError * _Nullable))completion {
+- (void)prepareForAppCheckWithCompletion:(nullable GIDAppCheckCompletion)completion {
+  if (completion) {
+    @synchronized (self.prepareCompletions) {
```

Comment:
I think this should be `@synchronized (self)` instead because other accesses to `self.prepareCompletions` are synchronized on `self`.